torch/_subclasses/fake_tensor.py

```diff
@@ -234,28 +233,14 @@ def invalidate_constant_aliases(self, tensor):
         del self.constant_storage_mapping[weak_st]
 
     def _get_memo(self, t):
-        if WeakIdRef(t) in self.tensor_memo:
-            out = self.tensor_memo[WeakIdRef(t)]
-            out._fix_weakref()
-            return out
-        return None
+        tid = self.meta_converter.describer.lookup_tensor.get(t)
+        if tid is None:
+            return None
+        return self.tensor_memo.get(tid)
 
     def set_tensor_memo(self, t, v):
-        th = WeakIdRef(t)
-
-        # hold a weak ref to self, otherwise it will be kept alive
-        # by the del_ten closure
-        self_weak_ref = weakref.ref(self)
-
-        def del_ten():
```

Comment:
Was it difficult to retain this behavior or was it intentional ? I read your comment in the summary, but I don't know the follow the case where either the fake tensor or real tensor dies and we wish to preserve the entry.

From a debugging perspective it's nice to remove the tensors from the dictionary when they are no longer needed

Reply:
Now that I am factoring through the id, I cannot drop the entries because someone may have held the int id and expect the entry to stay.

To be clear, we ARE removing entries from the dictionary when they are not needed. When a real tensor dies, we remove it from `tensor_lookup`. When a fake tensor dies, we remove it from `tensor_memo`. This happens automatically with WeakKey/ValueDictionary.